backend/src/controller/chatController.ts

```diff
@@ -187,38 +192,79 @@ async function handleChatToGPT(req: OpenAiChatRequest, res: Response) {
 				chatModel
 			);
 		}
+	} catch (error) {
+		const errorMessage =
+			error instanceof Error ? error.message : 'Failed to get chatGPT reply';
+		handleErrorGettingReply(req, res, currentLevel, chatResponse, errorMessage);
+		return;
+	}
 
-		// if the reply was blocked then add it to the chat history
-		if (chatResponse.defenceReport.isBlocked) {
-			req.session.levelState[currentLevel].chatHistory.push({
-				completion: null,
-				chatMessageType: CHAT_MESSAGE_TYPE.BOT_BLOCKED,
-				infoMessage: chatResponse.defenceReport.blockedReason,
-			});
-		} else if (!chatResponse.reply || chatResponse.reply === '') {
-			throw new Error('Failed to get chatGPT reply');
-		}
+	// more error handling
+	if (chatResponse.openAIErrorMessage) {
+		handleErrorGettingReply(
+			req,
+			res,
+			currentLevel,
+			chatResponse,
+			simplifyOpenAIErrorMessage(chatResponse.openAIErrorMessage)
+		);
+		return;
+	} else if (!chatResponse.reply || chatResponse.reply === '') {
+		handleErrorGettingReply(
```

Comment:
i think there shouldn't be an error here as we are have responses which are blocked by defences set to null. treating as an error then results in 

![image](https://github.com/ScottLogic/prompt-injection/assets/118981273/d1ae4c08-c264-4ba4-b3de-beb2877fc5e6) (char limit set to 20)

Reply:
good spot! Fixed

Reply:
I'll add a unit test for this case too